### PR TITLE
Upload notification 

### LINKED ISSUE #2097: [Files] Uploading a large number of files in separate directories causes a very large number of unnecessary notifications to pop up.
**Describe the bug**
When trying to upload various directories and their contained files at the same time, a notification will pop up for every directory being uploaded, taking up the whole height of the screen.

**To Reproduce**
1. Upload a large number of files in separate directories.

**Expected behavior**
A single notification pertaining to the single transfer of files to Files should be shown to the user, rather than a separate notification for each directory.

**Screenshots**
NotificationsUploading
NotificationsUploaded

**Desktop (please complete the following information):**
 - OS: Windows 10
 - Browser: Firefox
 - Version: 99.0.1

### PR DESCRIPTION
**Pull request by FSM1**
closes #2097

### CODE REVIEW

File: packages/files-ui/src/Contexts/FilesContext.tsx [hunk 1]
```diff
@@ -350,14 +350,14 @@ const FilesProvider = ({ children }: FilesContextProps) => {
...
-  const uploadFiles = useCallback(async (bucket: BucketKeyPermission, files: File[], path: string) => {
-    const hasOversizedFile = files.some(file => file.size > MAX_FILE_SIZE)
-    if (hasOversizedFile) {
-      addToast({
-        title: t`We can't encrypt files larger than 2GB. Some items will not be uploaded`,
-        type: "error"
-      })
-    }
+  const uploadFiles = useCallback(async (bucket: BucketKeyPermission, files: FileWithPath[], rootUploadPath: string) => {
+    // const hasOversizedFile = files.some(file => file.size > MAX_FILE_SIZE)
```
Comment (Tbaut) [inline packages/files-ui/src/Contexts/FilesContext.tsx]: Is this supposed to stay commented out?

File: packages/files-ui/src/Contexts/FilesContext.tsx [hunk 2]
```diff
@@ -377,20 +377,33 @@ const FilesProvider = ({ children }: FilesContextProps) => {
...
-        cancelToken
-      )
+      const paths = [...new Set(files.map(f => getParentPathFromFilePath(f.path)))]
+      const totalUploadSize = files.reduce((sum, f) => sum += f.size, 0)
+
+      let uploadedSize = 0
+      for (const path of paths) {
+        const filesToUpload = files.filter((f => getParentPathFromFilePath(f.path) === path))
+        const batchSize = filesToUpload.reduce((sum, f) => sum += f.size, 0)
+        console.log(`Uploading ${filesToUpload.length} files to ${path}`)
```
Comment (Tbaut) [inline packages/files-ui/src/Contexts/FilesContext.tsx:387]: stray log

File: packages/files-ui/src/Contexts/FilesContext.tsx [hunk 3]
```diff
@@ -377,20 +377,33 @@ const FilesProvider = ({ children }: FilesContextProps) => {
...
+            updateToast(toastId, {
+              ...toastParams,
+              progress: Math.ceil(
+                ((progressEvent.loaded + uploadedSize) / totalUploadSize) * 100
+              )
+            })
+          },
+          cancelToken
+        )
+        console.log(`Finished uploading ${filesToUpload.length} files to ${path}`)
```
Comment (Tbaut) [inline packages/files-ui/src/Contexts/FilesContext.tsx]: stray log
```suggestion
```

**Review by Tbaut — APPROVED** [review]
Such a beautiful refactor. Agreed that the `for .. of`here is better than the `.reduce` with some weird await :)

Working well, I tested with deeply nested directories without issue.

File: packages/files-ui/src/Contexts/FilesContext.tsx (hunk 2 above)
Comment (Tbaut) [inline packages/files-ui/src/Contexts/FilesContext.tsx]:
```suggestion
```

File: packages/files-ui/src/Utils/getFilesFromDataTransferItems.ts [hunk 4]
```diff
@@ -75,4 +77,6 @@ const getFilesFromDataTransferItems = async (dataTransferItems: DataTransferItem
   return files
 }
 
+
+
```
Comment (Tbaut) [inline packages/files-ui/src/Utils/getFilesFromDataTransferItems.ts]:
```suggestion
```

**Review by juans-chainsafe — APPROVED** [review]
Working as expected in all browsers, amazing job!

**Review by tanmoyAtb — APPROVED** [review]
Looks clean and smooth !

**Pull request merged**